cocos/core/root.ts

```diff
@@ -224,11 +220,15 @@ export class Root {
      */
     public initialize (info: IRootInfo): boolean {
 
-        if (!this._device.mainWindow) {
-            return false;
-        }
-
-        this._mainWindow = this._device.mainWindow;
+        this._mainWindow = this.createWindow({
+            title: 'rootMainWindow',
+            width: this._device.width,
+            height: this._device.height,
+            renderPassInfo: {
+                colorAttachments: [ new GFXColorAttachment() ],
+                depthStencilAttachment: new GFXDepthStencilAttachment(),
+            },
```

Comment:
Seems UI also have a root ? Does that mean it will create useless color attachment and depth stencil attachment

Reply:
UI only have a unnecessary render-scene, IMHO. The root here is maintained globally as a singleton, so no redundancy here.